Source/CountDown.spoon/init.lua

```diff
@@ -90,42 +280,92 @@ end
 ---  * None
 
 function obj:pauseOrResume()
-    if obj.timer then
-        if obj.timer:running() then
-            obj.timer:stop()
-        else
-            obj.timer:start()
-        end
+   -- if the timer is paused, we need to offset
+   -- the starting time for as long as the timer is paused
+
+   if obj.timer and obj.timerBar then
+      if obj.timer:running() then
+         obj.pausedAt = obj:time_absolute_seconds()
+         obj.timer:stop()
+         obj.timerBar:stop()
+
+         obj:menu_items(true)
+
+      else
+         -- offset the starting time by as many seconds as we were paused
+         local pausedSeconds = obj:time_absolute_seconds() - obj.pausedAt
+         obj.startTime = obj.startTime + pausedSeconds
+         obj.pausedAt = nil
+         obj.timer:start()
+         obj.timerBar:start()
+
+         obj:menu_items(false)
+      end
     end
 end
 
+--- CountDown:notify()
+--- Method
+---    Notify user time has expired
+---
+--- Parameters:
+---  * None
+
+function obj:notify()
+   strTime = "Time is " .. os.date("%X")
+   message = "Time (" .. obj.lenMin .. " mins) is up!"
+   
+   hs.notify.new({
+         title = message,
+         informativeText = strTime,
+         hasActionButton = true,
+         withdrawAfter = 100
+   }):soundName(obj.sound):send()
+
+   hs.alert.show(message, nil, hs.screen.mainScreen(), obj.alertLenMin)
+   hs.dialog.blockAlert(message, strTime)
+end
+
 --- CountDown:setProgress(progress)
 --- Method
 --- Set the progress of visual indicator to `progress`.
 ---
 --- Parameters:
----  * progress - an number specifying the value of progress (0.0 - 1.0)
-
-function obj:setProgress(progress, notifystr)
+---  * progress - a number specifying the value of progress (0.0 - 1.0)
+---
+function obj:setProgress(progress)
     if obj.canvas:frame().h == 0 then
         -- Make the canvas actully visible
-        local mainScreen = hs.screen.mainScreen()
+        local mainScreen = hs.screen.primaryScreen()
         local mainRes = mainScreen:fullFrame()
```

Comment:
wrong variable naming
```suggestion
        local primaryScreen = hs.screen.primaryScreen()
        local mainRes = primaryScreen:fullFrame()
```